### DISCUSSION TITLE
Customized button style

### DISCUSSION THREAD

**Discussion by frankfzhao** [twilio-labs/paste#1549] [Office hours] [answered]
### Get feedback on a customized button style used in a project for Marketing Campaign product.

**Customers**
The tool is used by Olivia (marketers) to create condition logics for dynamic segments.

**Use cases**
When customers use the product, they need to switch the logic between conditions between and / or and through testing, most customers (4/4) thinks that the and / or work best as a dropdown button to quickly switch between 2 logics.

**Design**
I have used a "badge" ish shape and `color-background-primary-lighter` as the background color for `And` button. Same shape but `color-background-new` as the background color for `Or` button 

View it here:
![And : OR customized button](https://user-images.githubusercontent.com/82524550/121624603-120c4600-ca2f-11eb-962e-1cd5c985fbd2.png)

Design prototype is [here](https://www.figma.com/proto/j5h6mhVWU9mf6dkEtUhQ8Z/And-or-V2?page-id=884%3A0&node-id=884%3A1&viewport=-637%2C-470%2C0.17196068167686462&scaling=min-zoom)

I would love some suggestions on the usage on the customized button style. Thanks

**Comment by richbachman** [comment]
Hi @frankfzhao. Thanks for sharing. I'll share with the team, so we can provide feedback before the next office hours.

**Comment by frankfzhao** [comment]
Adding some more here. I would also like to talk about the hover state for icons. Same screenshot above, I am wondering if we have something in DS that supports hovering state for the icons in the card. (Or probably I should use the icon only component[Not sure which style to use though])

**Comment by serifluous** [comment]
Hey @frankfzhao - Here's the video from office hours: https://www.loom.com/share/e25a286e90ab4584bd42bb9ae424d17c. The password is posted in the #help-design-system channel. Thanks again for coming to office hours!

**Comment by SiTaggart** [comment, marked as answer]
Some extra thoughts here after the team got together discussion the and/or logic gates.

## Component identity

Starting with the control itself 

![image](https://user-images.githubusercontent.com/368249/123486029-28cea300-d5c0-11eb-90dd-c24b628bb0b0.png)

What did come up most is what this component actually is, especially when we are talking about "button styles". With that in mind we believe the intention is that the control is a menu button that shows a menu of selectable items. The potential wrench that I threw in was that menu buttons shouldn't change their text upon selection. Menus perform actions, they don't set a value or persist state.

For that, you're looking at a [Combobox](https://paste.twilio.design/components/combobox).

Alternatively if you've only got and or or, you might consider some sort of switch control, which we admittedly don't have yet.

## Grouping and association

The next part comes as the current design is pretty heavily reliant on visual associate and visual meaning to communicate some pretty important relationships. The canvas nature of it might lend itself to being somewhat tricky to provide non-visual structure and relationships to users of assistive technology.

With that we were wondering if you'd be interested in exploring something a little more structured and hierarchical, which would then lend itself better to supporting the use of a Combobox for the logic gate. 

As prior art we're thinking of this (which I can't remember the name of the product) 

![image](https://user-images.githubusercontent.com/368249/123488372-a8f70780-d5c4-11eb-97eb-9b19b5cd45f5.png)

or this great example from Salesforce 

![image](https://user-images.githubusercontent.com/368249/123488444-d5128880-d5c4-11eb-892b-e9e9e6153238.png)

https://www.lightningdesignsystem.com/components/expression/

Especially with the Salesforce example, the non-visual communication of state, relationships, hierarchy and meaning is really clear. We spent quite a lot of time on it.

Other advantages might include its ability to scale pretty well to very complex expressions, with having to build some sort of zoom in and out functionality, and generally being easier to build for the engineering team, if that's a consideration.

## In summary

I don't think the addition of a new token would be wise right now, given that I don't think that control should be a button, and there might be better ways to communicate groupings, relationships and structure to non-sighted users.
